### PR TITLE
fix compose build integration

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes: It was rebuilding the image on the deploy if there was a stack

## Proposed changes
- If run okteto build with a compose it will run and include the host volumes on the built image
- If run okteto deploy it will build the images at the beginning and not while deploying stack

### CODE REVIEW

File: pkg/model/const.go [hunk 1]
```diff
@@ -317,4 +317,7 @@ const (
 
 	// OktetoDefaultImageTag default tag assigned to image to build
 	OktetoDefaultImageTag = "okteto"
+
+	// OktetoImageTagWithVolumes is the tag assigned to an image with volume mounts
+	OktetoImageTagWithVolumes = "okteto-with-volume-mounts"
```
Comment (pchico83) [inline pkg/model/const.go:322]: can we build the same tag in both cases? I think it would simplify the code

File: pkg/model/manifest.go [hunk 2]
```diff
@@ -236,15 +236,15 @@ func GetManifestV2(manifestPath string) (*Manifest, error) {
...
-			inferredManifest.mergeWithOktetoManifest(devManifest)
-		} else if inferredManifest.Type == StackType {
+		if inferredManifest.Type == StackType {
 			inferredManifest, err = inferredManifest.InferFromStack()
 			if err != nil {
 				return nil, err
 			}
-			//TODO: uncomment
-			// inferredManifest.Deploy.Compose.Stack.Endpoints = devManifest.Deploy.Endpoints
+			inferredManifest.Deploy.Compose.Stack.Endpoints = devManifest.Deploy.Endpoints
```
Comment (pchico83) [inline pkg/model/manifest.go:244]: I am getting a segmentation fault error here when running `okteto build` in this repo:
https://github.com/pchico83/compose-getting-started

**Review by pchico83 — COMMENTED** [review]
If I run `okteto build` and `okteto deploy` on this repo:
https://github.com/pchico83/compose-getting-started

The image for `vote` is registry.cloud.okteto.net/pchico83/compose-getting-started-vote:okteto-with-volume-mounts, but it should be the tag based on the `sha256:56df26d0b78b786c248d633251ca` to make sure the containers are redeployed if the image has changed.

File: pkg/model/const.go (hunk 1 above)
Comment (jLopezbarb) [reply pkg/model/const.go:322]: I don't think it's a good idea because of this use case:
1.  User run `okteto deploy` for the first time 
2. It builds the first image with the tag `okteto`
3. It fails on including the volumes
4. User runs again `okteto deploy` and it won't be built because the tag exists (right now we are checking if the tag with the volumes exists)

Comment (pchico83) [reply pkg/model/const.go:322]: Makes sense

File: pkg/model/manifest.go (hunk 2 above)
Comment (jLopezbarb) [reply pkg/model/manifest.go:244]: I can't repro it, are you using any flag on the build command?

Comment (pchico83) [reply pkg/model/manifest.go:244]: Nope, just `okteto build` or `okteto deploy`. Let me try again

**Review by pchico83 — APPROVED** [review]
